### PR TITLE
Adds the UV-T Skink scouting drone

### PR DESCRIPTION
**Pull request by blackdav123**
## About The Pull Request
Adds the UV-T Skink, a tiny and fragile unmanned vehicle used for scouting enemy positions.
image

While lacking weaponry, it is easily portable and cheap to replace.
image

Engineers can get it and a remote for 5 points, or it is available in req for 50 points.

Other changes:

- The unmanned vehicle remote is now a tiny item instead of normal.
## Why It's Good For The Game
Engineers dont have very many interesting things to buy below 10 points. This provides a low impact item that is still interesting to use and can enable better team play.
## Changelog
:cl:
Add: Added UV-T Skink scouting drone.
Balance: The unmanned vehicle remote is now a tiny item.
/:cl:

### CODE REVIEW

**Comment by iforgotmeword** [comment]
Does it have a light source?

**Comment by ilikesaying** [comment]
> Does it have a light source?

It has night vision according to the code

File: code/game/objects/items/radio/detpack.dm [hunk 1]
```diff
@@ -255,7 +255,7 @@
 		set_frequency(signaler.frequency)
 		to_chat(user, "You transfer the frequency and code of [signaler] to [src].")
 		return
-	if(istype(target, /obj/item) || istype(target, /mob))
+	if(istype(target, /obj/item) || istype(target, /mob) || istype(target, /obj/vehicle/unmanned))
```
Comment (Lumipharon) [inline code/game/objects/items/radio/detpack.dm:258]: whytho?

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -725,6 +725,10 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 /obj/vehicle/unmanned/projectile_hit(obj/projectile/proj, cardinal_move, uncrossing)
 	if(proj.shot_from == src)
 		return FALSE
+	if(src == proj.original_target)
+		return TRUE
+	if(allow_pass_flags & PASS_PROJECTILE)
+		return FALSE
```
Comment (Lumipharon) [inline code/modules/projectiles/projectile.dm:731]: Cursed way of doing what parent already does. I have this fixed in another pr though

File: code/modules/vehicles/unmanned/deployable_vehicles.dm [hunk 3]
```diff
@@ -65,3 +65,44 @@
...
+	deployable_item = /obj/vehicle/unmanned/deployable/tiny
+
+/obj/vehicle/unmanned/deployable/tiny
+	name = "UV-T Skink"
+	icon_state = "tiny_uv"
+	density = FALSE
+	move_delay = 1.3
+	hud_possible = list(MACHINE_HEALTH_HUD)
+	atom_flags = NONE
+	soft_armor = list(MELEE = 25, BULLET = 25, LASER = 25, ENERGY = 100, BOMB = 25, BIO = 100, FIRE = 25, ACID = 25)
```
Comment (Lumipharon) [inline code/modules/vehicles/unmanned/deployable_vehicles.dm:84]: pls no mystery 100 energy armour

File: code/modules/vehicles/unmanned/unmanned_vehicle_remote.dm [hunk 4]
```diff
@@ -3,6 +3,7 @@
 	desc = "Used to control an unmanned vehicle.<br>Tap the vehicle you want to control with the controller to link it."
 	icon = 'icons/obj/device.dmi'
 	icon_state = "multitool2"
+	w_class = WEIGHT_CLASS_TINY
```
Comment (Lumipharon) [inline code/modules/vehicles/unmanned/unmanned_vehicle_remote.dm:6]: why's it tiny? Everything else like this is small

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by Mannybrado** [comment]
how fast is it doe

File: code/game/objects/items/radio/detpack.dm (hunk 1 above)
Comment (blackdav123) [reply code/game/objects/items/radio/detpack.dm:258]: i remembered this being a problem in the past for other unmanned and dont want it on this one especially, i can make a flag to put on this if you'd prefer

File: code/modules/vehicles/unmanned/unmanned_vehicle_remote.dm (hunk 4 above)
Comment (blackdav123) [reply code/modules/vehicles/unmanned/unmanned_vehicle_remote.dm:6]: thought it would be nice to have it on helmet but i'll change it

**Comment by blackdav123** [comment]
> how fast is it doe

slightly faster than a runner off of weeds

**Comment by DrAmazing343** [comment]
Just to be sure, this won't proc Gargoyles, right? Feels like the perfect kind of thing to scout out Xeno hardpoints to plan an attack which is gameplay we basically never ever see in the very unga and reactive meta of today.

**Comment by blackdav123** [comment]
> Just to be sure, this won't proc Gargoyles, right? Feels like the perfect kind of thing to scout out Xeno hardpoints to plan an attack which is gameplay we basically never ever see in the very unga and reactive meta of today.

seems it does, i'll try to make it not do that

**Comment by Looking4bros** [comment]
does it trigger acid wells? i'd think an acid well would spell doom for such a little device.

**Comment by blackdav123** [comment]
> does it trigger acid wells? i'd think an acid well would spell doom for such a little device.

it does not

**Comment by SimplySmiley123** [comment]
Not gonna lie this would be awesome.
Doubt it's gonna be bought much by engineers, especially with their vendor points, but I can imagine squaddies having some fun with it every once in a while.

**Comment by despong** [comment]
so we're in copper now
will we be able to throw it to deploy it instantly like in r6 or would you still have to bring it out inhand and then ctrl click the ground like the rest of the ugvs

**Comment by blackdav123** [comment]
> Can it move under xenos like a larva?

yes

**Comment by Admiral-Gigaton** [comment]
damn, not letting me duct tape an IED to my drones is such a 1984, you monster! Guerilla warfare will never be the same!

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**